app/models/postcode_local_restriction_search.rb

```diff
@@ -0,0 +1,53 @@
+class PostcodeLocalRestrictionSearch
+  UK_POSTCODE_PATTERN = %r{
+    \A
+    ([Gg][Ii][Rr] 0[Aa]{2}) # Unusual postcodes
+    |
+    (
+      # Outward code
+      (([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))
+      \s?
+      [0-9][A-Za-z]{2} # Inward code
+    )
+    \Z
+  }x.freeze
+
+  attr_reader :query, :normalised_postcode
+  delegate :no_information?, to: :location_lookup
+
+  def initialize(query)
+    @query = query
+    # Use the uk_postcode gem to potentially transpose O/0 and I/1.
+    @normalised_postcode = UKPostcode.parse(
+      query.gsub(/[^a-z0-9 ]/i, "").strip,
+    ).to_s
+  end
+
+  def error_code
+    return "postcodeLeftBlank" if query.empty?
```

Comment:
I think it's ok that this expects the input to be a string - we don't have a use case for passing this a nil value. 

If we consider that it could be nil it'll require a bunch of considerations. One you've pointed in your next comment, the use of gsub and strip, and beyond that we've got a match method and both UKPostocode and LocationLookupService will exception without a string. 

If there's a reason to be genuinely concerned of usage with nil rather than string the simplest approach would be to have the constructor do `@input = input || ""` but this doesn't seem that necessary to me as you shouldn't initialise this object if you haven't received a search request.